decorator.rb

```diff
@@ -0,0 +1,29 @@
+require_relative './nameable'
+
+# Decorator class
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# CapitalizeDecorator class
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+# TrimmerDecorator class
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```

Comment:
- [ ] It would be very nice to check if the name characters do not exceed the actual number we want before performing trim, just for the sake of the app performance, like the following:
```suggestion
    @nameable.correct_name.length > 10 ? @nameable.correct_name[0, 10] : @nameable.correct_name
```